atox/src/main/kotlin/ui/contactlist/ContactListFragment.kt

```diff
@@ -182,19 +175,18 @@ class ContactListFragment :
         }
 
         arguments?.getString(ARG_SHARE)?.let { share ->
-            shareDialog = ReceiveShareDialog(
-                requireContext(),
-                (binding.contactList.adapter as ContactAdapter).contacts,
+            val shareDialog = ReceiveShareDialogFragment(
+                viewModel.contacts,
                 share,
-            ) {
-                viewModel.onShareText(share, it)
-                openChat(it)
-            }
-            shareDialog?.setOnDismissListener {
-                shareDialog = null
-                arguments?.remove(ARG_SHARE)
-            }
-            shareDialog?.show()
+                onContactSelected = {
+                    viewModel.onShareText(share, it)
+                    openChat(it)
+                },
+                onDialogDismissed = {
+                    arguments?.remove(ARG_SHARE)
+                },
+            )
+            shareDialog.show(childFragmentManager, null)
```

Comment:
Shouldn't this be chained?

Reply:
You're right! Missed updating that when removing the bits in the middle. :P